File: Grind75/Week-2/169.majority-element.cpp
#include <bits/stdc++.h>
using namespace std;
/*
 * @lc app=leetcode id=169 lang=cpp
 *
 * [169] Majority Element
 */

// @lc code=start
class Solution
{
public:
    int majorityElement(vector<int> &nums)
    {
        int ele = nums[0];
        int count = 1;

        for (int i = 1; i < nums.size(); i++)
        {
            if (!count)
            {
                ele = nums[i];
                count++;
            }
            else if (ele == nums[i])
            {
                count++;
            }
            else if (count > nums.size() / 2)
            {
                return ele;
            }
            else
            {
                count--;
            }
        }
        return ele;
    }
};
// @lc code=end

File: Java/Grind75/Week2/169.majority-element.java
package Grind75.Week2;
/*
 * @lc app=leetcode id=169 lang=java
 *
 * [169] Majority Element
 */

// @lc code=start
class Solution {
    public int majorityElement(int[] nums) {
        int majorityElement = 0, count = 0;
        for (int i = 0; i < nums.length; i++) {
            if (count == 0)
                majorityElement = nums[i];
            if (nums[i] == majorityElement)
                count++;
            else
                count--;
        }
        return majorityElement;
    }
}
// @lc code=end